### PR TITLE
Storage Tweak/Backpack Buff

### PR DESCRIPTION
**Pull request by Surrealaser**
-Backpack storage increased to 30.

-Standard and Engineer Backpacks can now store M56D and sentry parts.

-Welderpack fuel capacity increased to 500, and given backpack space so more engineers actually wear them.

-M56D and sentry crates can now store their parts.

-Medic backpack now has an integrated defibrillator recharge unit, complete with a high capacity power cell.

-Dragging a defibrillator onto the medic backpack will transfer charge from the recharge unit to the defibrillator.

-Dragging a power cell onto the medic backpack will replace the currently loaded power cell for the recharge unit.

### CODE REVIEW

File: code/game/objects/items/storage/backpack.dm [hunk 1]
```diff
@@ -444,8 +466,7 @@
 	name = "\improper USCM technician welderpack"
 	desc = "A specialized backpack worn by USCM technicians. It carries a fueltank for quick welder refueling and use,"
 	icon_state = "engineerpack"
-	var/max_fuel = 260
-	max_storage_space = 15
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]: This buff is not necessary.

File: code/modules/cm_marines/equipment/sentries.dm [hunk 2]
```diff
@@ -19,7 +19,12 @@
 	icon_state = "sentry_case"
 	w_class = 5
 	storage_slots = 6
-	can_hold = list() //Nada. Once you take the stuff out it doesn't fit back in.
```
Comment (Rohesie) [inline code/modules/cm_marines/equipment/sentries.dm]: Have you tested the effect of this?

File: code/game/gamemodes/colonialmarines/halloween2016.dm [hunk 3]
```diff
@@ -773,7 +773,7 @@
 					if(12 to 15) //SADAR. the most popular choice, but also pretty damn deadly.
 						H.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/marine(H), WEAR_HEAD)
 						H.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine(H), WEAR_JACKET)
-						H.equip_to_slot_or_del(new /obj/item/storage/backpack/marine(H), WEAR_BACK)
+						H.equip_to_slot_or_del(new /obj/item/storage/backpack/marine/standard(H), WEAR_BACK)
```
Comment (Rohesie) [inline code/game/gamemodes/colonialmarines/halloween2016.dm]: This is a both unneeded and undocumented change.

File: code/game/objects/items/storage/backpack.dm [hunk 4]
```diff
@@ -292,15 +289,40 @@
...
+					"/obj/item/device/turret_top",
+					"/obj/item/device/turret_sensor",
+					"/obj/item/ammo_magazine/sentry",
+					)
+
 /obj/item/storage/backpack/marine/medic
 	name = "\improper USCM medic backpack"
 	desc = "The standard-issue backpack worn by USCM medics."
 	icon_state = "marinepackm"
+	bypass_w_limit = list("/obj/item/device/m56d_gun",
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]: Medic backpacks don't need this.

File: code/game/objects/items/storage/backpack.dm [hunk 5]
```diff
@@ -292,15 +289,40 @@
 			select_gamemode_skin(type)
 		..()
 
+/obj/item/storage/backpack/marine/standard
+	name = "\improper lightweight IMP backpack"
+	desc = "The standard-issue pack of the USCM forces. Designed to slug gear into the battlefield."
+	bypass_w_limit = list("/obj/item/device/m56d_gun",
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]: Standard backpacks don't need this.

File: code/game/objects/items/storage/backpack.dm (hunk 1 above)
Comment (Surrealaser) [reply code/game/objects/items/storage/backpack.dm]: Which is why the welderpack is such a popular choice, right?

File: code/modules/cm_marines/equipment/sentries.dm (hunk 2 above)
Comment (Surrealaser) [reply code/modules/cm_marines/equipment/sentries.dm]: Yes, it means you can actually put stuff back in.

File: code/game/gamemodes/colonialmarines/halloween2016.dm (hunk 3 above)
Comment (Surrealaser) [reply code/game/gamemodes/colonialmarines/halloween2016.dm]: If it's not done, they'll be stuck with the shitty base backpack which I'm essentially replacing.

File: code/game/objects/items/storage/backpack.dm (hunk 4 above)
Comment (Surrealaser) [reply code/game/objects/items/storage/backpack.dm]: All backpacks should have this capability.

File: code/game/objects/items/storage/backpack.dm (hunk 5 above)
Comment (Surrealaser) [reply code/game/objects/items/storage/backpack.dm]: All backpacks should have this capability.

File: code/game/objects/items/storage/backpack.dm (hunk 1 above)
Comment (Rohesie) [reply code/game/objects/items/storage/backpack.dm]: Yea, welderpack is kind of the standard engie gear. I see few engies that know what they are doing without one.

File: code/modules/cm_marines/equipment/sentries.dm (hunk 2 above)
Comment (Rohesie) [reply code/modules/cm_marines/equipment/sentries.dm]: What about other kind of stuff?

File: code/game/gamemodes/colonialmarines/halloween2016.dm (hunk 3 above)
Comment (Rohesie) [reply code/game/gamemodes/colonialmarines/halloween2016.dm]: The satchel is not shitty, and it's the standard one.

File: code/game/objects/items/storage/backpack.dm (hunk 4 above)
Comment (Rohesie) [reply code/game/objects/items/storage/backpack.dm]: Not really.

File: code/game/objects/items/storage/backpack.dm (hunk 5 above)
Comment (Rohesie) [reply code/game/objects/items/storage/backpack.dm]: Not really.

File: code/game/objects/items/storage/backpack.dm (hunk 1 above)
Comment (Surrealaser) [reply code/game/objects/items/storage/backpack.dm]: They'll take it along as a fuel repository, but not as something they actually wear.

File: code/modules/cm_marines/equipment/sentries.dm (hunk 2 above)
Comment (Surrealaser) [reply code/modules/cm_marines/equipment/sentries.dm]: Who cares? It's a box. I don't see why it should magically preclude stuff from going on.

File: code/game/gamemodes/colonialmarines/halloween2016.dm (hunk 3 above)
Comment (Surrealaser) [reply code/game/gamemodes/colonialmarines/halloween2016.dm]: No, the shitty base _backpack_ which is the _root_ of the other backpacks/satchels.

File: code/game/objects/items/storage/backpack.dm (hunk 4 above)
Comment (Surrealaser) [reply code/game/objects/items/storage/backpack.dm]: Yes. No need for all these ridiculous arbitrary differences, and it's one at least semi-compelling reason to have a backpack; sure it's generally not going to be a thing for medics, but PFCs are often on M56D duty at the least.

File: code/game/gamemodes/colonialmarines/halloween2016.dm (hunk 3 above)
Comment (Rohesie) [reply code/game/gamemodes/colonialmarines/halloween2016.dm]: Alright.

File: code/modules/cm_marines/equipment/sentries.dm (hunk 2 above)
Comment (Rohesie) [reply code/modules/cm_marines/equipment/sentries.dm]: Alright.

File: code/game/objects/items/storage/backpack.dm (hunk 1 above)
Comment (Rohesie) [reply code/game/objects/items/storage/backpack.dm]: It's a satchel-sized backpack with fuel in it. There are trade-offs.

File: code/game/objects/items/storage/backpack.dm (hunk 5 above)
Comment (Rohesie) [reply code/game/objects/items/storage/backpack.dm]: The engineering backpack provides an engineer-relevant function. That is good. The rest don't.

File: code/game/objects/items/storage/backpack.dm (hunk 4 above)
Comment (Rohesie) [reply code/game/objects/items/storage/backpack.dm]: Think about adding something medical related for the medic backpacks instead (no large containers such as medkits, please).

Comment (Surrealaser) [reply code/game/objects/items/storage/backpack.dm]: Added a defibrillator recharge unit.

**Comment by Rohesie** [comment]
Closing until the requested changes are done.

File: code/game/objects/items/storage/backpack.dm [hunk 6]
```diff
@@ -292,15 +289,100 @@
...
+	desc = "The standard-issue pack of the USCM forces. Designed to slug gear into the battlefield."
+
 /obj/item/storage/backpack/marine/medic
 	name = "\improper USCM medic backpack"
 	desc = "The standard-issue backpack worn by USCM medics."
 	icon_state = "marinepackm"
+	var/obj/item/cell/high/cell //Starts with a high capacity energy cell.
+
+/obj/item/storage/backpack/marine/medic/New()
+	..()
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]:
```suggestion
	. = ..()
```

File: code/game/objects/items/storage/backpack.dm [hunk 7]
```diff
@@ -292,15 +289,100 @@
...
+		else
+			warning = "<span class='warning'>[src]'s defibrillator recharge unit buzzes a warning, as its battery is completely depleted of charge. "
+	else
+		playsound(src, 'sound/machines/ping.ogg', 25, 1)
+		warning = "<span class='notice'>[src]'s defibrillator recharge unit cheerfully pings as it successfully recharges the defibrillator. "
+	cell.charge -= min(cell.charge, amount)
+	if(mention_charge)
+		to_chat(user, "<span class='notice'>[warning]<b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+	update_icon()
+	..()
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]:
```suggestion
	return ..()
```

File: code/game/objects/items/storage/backpack.dm [hunk 8]
```diff
@@ -292,15 +289,100 @@
...
+
+/obj/item/storage/backpack/marine/medic/examine(mob/user)
+	..()
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]:
```suggestion
	. = ..()
```

File: code/game/objects/items/storage/backpack.dm [hunk 9]
```diff
@@ -292,15 +289,100 @@
...
+	if(cell)
+		to_chat(user, "<span class='notice'>Its defibrillator recharge unit has a loaded power cell and its readout counter is active. <b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+	else
+		to_chat(user, "<span class='warning'>Its defibrillator recharge unit does not have a power cell installed!</span>")
+
+/obj/item/storage/backpack/marine/medic/update_icon()
+	icon_state = initial(icon_state)
+	if(cell && cell.charge)
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]:
```suggestion
	if(cell?.charge)
```

File: code/game/objects/items/storage/backpack.dm [hunk 10]
```diff
@@ -292,15 +289,100 @@
...
+			if(!cell)
+				replace_install = "You install a cell in [src]'s defibrillator recharge unit."
+			else
+				cell.updateicon()
+				user.put_in_hands(cell)
+			cell = W
+			to_chat(user, "<span class='notice'>[replace_install] <b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+			playsound(user, 'sound/weapons/gun_rifle_reload.ogg', 25, 1, 5)
+			update_icon()
+	..()
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]:
```suggestion
	return ..()
```

**Comment by Rohesie** [comment]
This will conflict with the turrets.

**Pull request merged**